sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/TransactionStatus.kt

```diff
@@ -2,8 +2,22 @@ package cash.z.ecc.android.sdk.internal.model
 
 import cash.z.ecc.android.sdk.model.BlockHeight
 
-interface TransactionStatus {
-    class TxidNotRecognized : TransactionStatus
-    class NotInMainChain : TransactionStatus
-    data class Mined(val height: BlockHeight) : TransactionStatus
-}
\ No newline at end of file
+sealed class TransactionStatus {
+    abstract fun toPrimitiveValue(): Long
+
+    data class Mined(val height: BlockHeight) : TransactionStatus() {
+        override fun toPrimitiveValue() = height.value
+    }
+
+    data object NotInMainChain : TransactionStatus() {
```

Comment:
Aha, `data object` is what I needed. When I initially tried using `sealed class`, I got compiler errors about the parent class having no data, which is why I went for the interface approach (there being no other obvious method to do this described in the Kotlin documentation).

Reply:
Yeah, I see. I changed that to `sealed class`.